fix fbxloader error

FishOrBear opened the pull request:
[Drawing1.zip](https://github.com/mrdoob/three.js/files/1225547/Drawing1.zip)
Fixed an error.
The error caused fbx not to be imported properly.

#9860

mrdoob commented on examples/js/loaders/FBXLoader.js:
```diff
@@ -4010,7 +4010,10 @@
 
 				var node = this.parseNode( reader, version );
-				if ( node !== null ) allNodes.add( node.name, node );
+				if ( node !== null ) 
```
How about reversing it?

```js
if ( node === null ) break;
allNodes.add( node.name, node );
```

FishOrBear replied:
nice.
Do i need to modify pr?

FishOrBear replied:
I have modified pr

takahirox commented:
Let me review if this fix is correct.

takahirox commented:
It doesn't seem this change is correct.
With Drawing1.fbx, this change makes `FBXLoader` break a loop at a certain point in the binary file but some data still exist after the point.
This means `FBXLoader` loses some data.

Probably "Unknown property type" isn't a root issue.
There seems being another root issue which causes it.
I'm taking a look closer...

FishOrBear commented:
3dmax before exporting
![image](https://user-images.githubusercontent.com/19372111/29345739-3e16e15e-8273-11e7-8365-9a60e6d7637b.png)
After import
[test.zip](https://github.com/mrdoob/three.js/files/1226960/test.zip)

![image](https://user-images.githubusercontent.com/19372111/29345757-571b90aa-8273-11e7-9d95-056391399410.png)

There is also a problem that causes the imported grid to appear incorrectly.

[Uploading testBox.zip…]()


@takahirox

takahirox commented:
Thanks for reporting. Would you please check if there's any existing similar/same issue threads and open another issue for that if there isn't?

FishOrBear commented:
@takahirox  
Find 2 related questions.

#11697  #11895

I copied the wrong instance file to  #11895

takahirox commented:
If I'm right, Drawing1.fbx includes two models...?
How was this file generated?

FishOrBear commented:
@takahirox 
This is the AutoCAD2016 derived fbx file.
If I quit in the middle, then the entity is loaded normally.

takahirox commented:
> If I quit in the middle

What do you mean by that?

FishOrBear commented:
Sorry, this is the result of Google translation, may be somewhat misleading.

That is, if i follow my code.
It can load the model correctly.

That is to say

If (null === node) // exit. (If I quit in the middle) (google translation error)


@takahirox

takahirox commented:
OK. Let me explain what I mentioned again.

FBX binary has 16-byte end-footer in the end of the file.

    0xf8 0x5a 0x8c 0x6a 0xde 0xf5 0xd9 0x7e 0xec 0xe9 0x0c 0xe3 0x75 0x8f 0x29 0x0b

But Drawing1.fbx has two footers, at address 0x4160 and 0x4440.
I suppose Drawing1.fbx includes two FBX data.
Probably you can load without any `FBXLoader` change if you cut the second data.

I'm really not sure if this "two (or more) combined data in one file" is proper FBX specification.

IMO We can consider to support it if
- There're (not a few) this type of existing files
- Other libs/viewers can load and render them correctly

FishOrBear commented:
This file is imported into u3d, or UE4, which can be imported correctly.

I do not know how to split the file.

takahirox commented:
Can you generate the ascii version file and share?
I'm wondering how we should handle the second data.
(Just ignore?)

FishOrBear commented:
@takahirox 
I do not know how to generate ascii files.
I think i can try fbxSDK

FishOrBear commented:
[Drawing1Ascii.zip](https://github.com/mrdoob/three.js/files/1236390/Drawing1Ascii.zip)
I tried using fbxConver to convert the file.


http://usa.autodesk.com/adsk/servlet/pc/item?siteID=123112&id=22694909

looeee commented:
@FishOrBear @takahirox  I've tested loading the original file (Drawing1.fbx) in a couple of ways, including importing it back into 3ds Max. In all cases it doesn't look like the original screenshot - it's just six planes making up a box. 

Additionally as you can see in the following screen shot, the planes are missing names - this is unusual (actually first time I've seen this in max, generally everything is named). 

![box](https://user-images.githubusercontent.com/5307958/29487745-83112122-8528-11e7-8861-23ec2b453335.jpg)


Could it be that the original export didn't work properly for some reason?

FishOrBear commented:
@looeee  
I guess this is not the reason.
You can try the following file.

[Drawing4.zip](https://github.com/mrdoob/three.js/files/1236821/Drawing4.zip)

takahirox commented:
Anyways, I want something proof which indicates this type of files is a proper FBX format. I'd never seen.

vidartf committed:
Docs: SpotLightShadow.camera tracks light

takahirox committed:
Add try-catch for GLTFBinaryExtension

takahirox committed:
Merge branch 'dev' into UpdateGLTFLoader

takahirox committed:
Update GLTFLoader documentation

looeee committed:
FBXLoader removed final JSDoc comments

Commit:
Clean unused and undeclared variable Bones

Commit:
Fix style issus

Commit:
Fix undefined parseASCII

Commit:
Fix unused baseDir

Commit:
Fix unused find

Commit:
Fix unused scope

Commit:
Fix unused param line

Commit:
Fix undefined XBoneInf

Commit:
Fix unused BoneDics, rootBone and keys

Mugen87 committed:
OBJLoader: Clean up

mrdoob committed:
Merge pull request #12551 from takahirox/GLTFLoaderAccessorNormalized

Support glTF accessor.normalized

looeee committed:
FBXLoader: remove duplicated arays from binary parser

mrdoob committed:
WebVR.js: Correctness.

mrdoob committed:
Merge pull request #12543 from takahirox/UpdateGLTFLoaderBufferErrorMessage

Generalize GLTFLoader buffer loading failure message

mrdoob committed:
Merge pull request #12558 from looeee/FBXLoader_binary_parser_duplicate_less_arrays

FBXLoader: remove duplicated arrays from binary parser

mrdoob committed:
Editor: Added project title.

mrdoob committed:
Merge pull request #12456 from donmccurdy/feat-imagebitmaploader-worker

ImageBitmapLoader: Support Firefox's partial createImageBitmap implementation.

mrdoob committed:
Merge pull request #12557 from Mugen87/dev7

OBJLoader: Clean up

mrdoob committed:
Merge pull request #12555 from looeee/FBXLoader_clean_comments

FBXLoader: Cleaned up comments

mrdoob committed:
Merge pull request #12556 from Itee/CleanXLoader

Clean unused and undeclared variable Bones

mrdoob committed:
Merge pull request #12512 from kchapelier/prwmloader-doc

PRWM Loader doc

mrdoob committed:
WebVRManager: Add missing window check. See #12461

mrdoob committed:
Merge pull request #12549 from Itee/unit_tests

Unit tests - New PR

mrdoob committed:
WebVR.js: Avoid glow effect.

mrdoob committed:
Fixed user height in webvr_vive examples.

mrdoob committed:
r88

looeee committed:
Docs: Buffergeometry clarify normal attribute

See #12559

paulmasson committed:
Typo

mrdoob committed:
Merge pull request #12561 from looeee/patch-11

Docs: Buffergeometry clarify normal attribute

mrdoob committed:
Merge pull request #12562 from paulmasson/patch-1

Typo

takahirox committed:
Add if ( onError ) check in GLTFLoader

takahirox committed:
Add (optional) for onError in GLTFLoader documentation

marcatec committed:
GLTFLoader: Allow alphaTest to be 0

makc committed:
support opacity in assimp2json loader

mrdoob committed:
Merge pull request #12566 from makc/patch-5

support opacity in assimp2json loader

mrdoob committed:
Merge pull request #12538 from takahirox/UpdateGLTFLoader

Add try-catch for GLTFBinaryExtension

mrdoob committed:
Merge pull request #12565 from marcatec/alphaTest

GLTFLoader: Allow alphaTest to be 0

vlucendo committed:
remove unused uniform

marcatec committed:
GLTFLoader: Replace for...in with classic for loops on arrays.

vidartf committed:
docs: grammar fix

marcatec committed:
Keep variable names for skinIndex and nodeIndex

JakeDluhy committed:
Fix doc typo

mrdoob committed:
Merge pull request #12573 from JakeDluhy/patch-1

Fix doc typo

mrdoob committed:
Merge pull request #12571 from marcatec/gltf-for-loops

GLTFLoader: Replace for...in with classic for loops on arrays.

mrdoob committed:
Merge pull request #12448 from vidartf/patch-1

Docs: Fix SpotLightShadow.camera.fov default value

mrdoob committed:
Editor: Removed counters.

keqingrong committed:
Update CanvasRenderer documentation

mrdoob committed:
Merge pull request #12576 from keqingrong/patch-1

Update CanvasRenderer documentation

Mugen87 committed:
Examples: Clean up

WestLangley committed:
Improved analyzer graphics; new music

donmccurdy committed:
GLTFLoader: Support sparse accessors.

mrdoob committed:
Merge pull request #12577 from Mugen87/dev4

Examples: Clean up

mrdoob committed:
Merge pull request #12570 from vlucendo/dev

remove unused uniform

Mugen87 committed:
Path: Removed PathPrototype

Mugen87 committed:
Path: Add constructor to definition

Commit:
Merge constructor and prototype in KeyframeTrack

mrdoob committed:
Merge pull request #12581 from WestLangley/dev-webaudio

Webaudio Analyzer: Improved graphics

mrdoob committed:
Merge pull request #12583 from Mugen87/dev4

Path: Removed PathPrototype

Commit:
Update error messages

Commit:
Update BooleanKeyframeTrack inheritance

Commit:
BooleanKeyframeTrack style

Commit:
Update ColorKeyframeTrack inheritance

Commit:
ColorKeyframeTrack style

Commit:
Update NumberKeyframeTrack inheritance

Commit:
NumberKeyframeTrack style

Commit:
Update QuaternionKeyframeTrack inheritance

Commit:
QuaternionKeyframeTrack style

Commit:
Update StringKeyframeTrack inheritance

Commit:
StringKeyframeTrack style

Commit:
Update VectorKeyframeTrack inheritance

Commit:
VectorKeyframeTrack style

Commit:
Remove reference of KeyframeTrackConstructor and KeyframeTrackPrototype in unit test

donmccurdy committed:
Docs: ColladaLoader now supports the new animation system.

mrdoob committed:
Merge pull request #12594 from mrdoob/donmccurdy-docs-collada-anim

Docs: ColladaLoader now supports the new animation system.

mrdoob committed:
Fixed webvr_vive_* examples.

mrdoob committed:
Merge branch 'master' into dev

mrdoob committed:
r89dev

mrdoob committed:
Math: Faster generateUUID. See #12432

mrdoob committed:
Updated builds.

Commit:
Prevent Command-Z from triggering tabs to open/close on Safari

mrdoob committed:
Editor: Changed default camera.near/far.

mrdoob committed:
Editor: Using degrees in geometry angle values.

cnspaha committed:
Check if mesh exists before working with it

There is the possibility that there isn't a 'mesh' attribute in the 'geometry' one so we should check if we found a mesh before trying to access its properties

Mugen87 committed:
WebVR: Update origin trial token

mrdoob committed:
Merge pull request #12607 from Mugen87/dev4

WebVR: Update origin trial token

mrdoob committed:
Math: Use toUpperCase() in the new generateUUID to match previous UUIDs. See #12432.

mrdoob committed:
Updated builds.

mrdoob committed:
Merge pull request #12584 from Itee/KeyFrameTrack

KeyframeTrack: Remove KeyframeTrackPrototype and KeyframeTrackConstructor

mrdoob committed:
Updated builds.

mrdoob committed:
Updated Rollup to ^0.51.0

mrdoob committed:
Updated builds.

mrdoob committed:
Merge pull request #12604 from TyLindberg/editor-safari-undo-fix

Editor: Prevent Command-Z from triggering tabs to open/close on Safari

cnspaha committed:
Refactoring code style, like suggested

https://github.com/mrdoob/three.js/pull/12606#issuecomment-343121765

cnspaha committed:
Added comment for unsupported tags

cnspaha committed:
ColladaLoader: Added possibility to manually set path

see https://github.com/mrdoob/three.js/issues/12609

cnspaha committed:
TDSLoader: Use path and manager for all Loaders

mrdoob committed:
Merge pull request #12606 from cnspaha/patch-5

ColladaLoader: Check if mesh exists before working with it

cnspaha committed:
Set local path like in GLTFLoader

kenkiku1021 committed:
fixed to read "serial" field in PDB files.

cnspaha committed:
Removed unnecessary line

cnspaha committed:
fixed tabs/spaces

Mugen87 committed:
WebVR: Added token to more examples

mrdoob committed:
Merge pull request #12624 from Mugen87/dev4

WebVR: Added origin trial token to all VR examples

Mugen87 committed:
Clean up

WestLangley committed:
Prevent needsUpdate true every frame

Mugen87 committed:
Examples: Added basic example for AnimationObjectGroup

mrdoob committed:
Merge pull request #12627 from Mugen87/dev4

Clean up

mrdoob committed:
Merge pull request #12628 from WestLangley/dev-materials_channels

Example: Prevent material.needsUpdate being true every frame

WestLangley committed:
Generalize Reflector geometry

Commit:
Examples: Manually manipulating morph targets on a model.

mrdoob committed:
Merge pull request #12629 from Mugen87/dev9

Examples: Added basic example for AnimationObjectGroup

WestLangley committed:
Generalize Reflector/Refractor geometry

mrdoob committed:
Merge pull request #12631 from WestLangley/dev-reflector_geometry

Generalize Reflector geometry

ngokevin committed:
remove unicode character from EllipseCurve.js

Mugen87 committed:
Examples: Clean up Refractor and Water2 (see #12631)

Mugen87 committed:
Examples: Improved Water

kenkiku1021 committed:
changed for coding style

Kos committed:
Docs: clarify requestAnimationFrame speed

It's not strictly true that requestAnimationFrame runs at 60fps - it depends on the device.
Assuming 60fps will make your app misbehave on computers with 75hz or 144hz screens.
This is an early tutorial, but let's at least give an early heads up that there's a gotcha here.

cnspaha committed:
TDSLoader: Fixed path logic

see https://github.com/mrdoob/three.js/pull/12615#issuecomment-343391055

cnspaha committed:
TDSLoader: fixed path

see https://github.com/mrdoob/three.js/pull/12615#discussion_r150508796

Mugen87 committed:
R88

looeee committed:
FBXLoader: added support for float color type

mrdoob committed:
Merge pull request #12641 from Kos/patch-1

Docs: clarify requestAnimationFrame speed

dmarcos committed:
Add the ability to specify an alternative Object3D to the camera to apply the pose coming from a VRDevice

mrdoob committed:
Merge pull request #12634 from ngokevin/unicodespace

remove unicode character from EllipseCurve.js

mrdoob committed:
Merge pull request #12620 from kenkiku1021/PDBLoader-fix

fixed to read "serial" field in PDB files.

mrdoob committed:
Merge pull request #12635 from Mugen87/dev9

Examples: Clean up Refractor and Water (see #12631)

mrdoob committed:
Improved webgl_mirror example.

mrdoob committed:
Improved ImageBitmapLoader example.

looeee committed:
Swtiched to color.setScalar

looeee committed:
FBXLoader: fix undefined children bug in parseAnimations

mrdoob committed:
Merge pull request #12653 from looeee/FBXLoader_parseAnimations_undefined_connection

FBXLoader: fix undefined connection bug in parseAnimations

looeee committed:
Merge branch 'dev' of https://github.com/mrdoob/three.js into FBXLoader_parseColor_fix

looeee committed:
removed redundant else statement

mrdoob committed:
Merge pull request #12648 from looeee/FBXLoader_parseColor_fix

FBXLoader: support for colors specified as single floats.

Mugen87 committed:
ShapeUtils: Use earcut for polygon triangulation (2nd attempt)

MagnuzBinder committed:
remove duplicate verticesNeedUpdate

MagnuzBinder committed:
fix typo Mixin => Mixing

MagnuzBinder committed:
fix index error in example

MagnuzBinder committed:
fix copy-paste error ParticleSystem => Particle

MagnuzBinder committed:
Remove unintelligible part of explanation

Not sure if the points are added to or replace present content of the curve?

MagnuzBinder committed:
consistent naming, hex => color

MagnuzBinder committed:
fix copy-paste error: models => images

MagnuzBinder committed:
fix copy-paste error: 2D => 3D

MagnuzBinder committed:
fixed copy-paste error: Vector3 => Box3

MagnuzBinder committed:
fix copy-paste error: 0, 1 or 2 => 0, 1, 2 or 3

MagnuzBinder committed:
remove duplicate text: domElement

MagnuzBinder committed:
fix typo: remove extra dot

MagnuzBinder committed:
use consistent language: hex => color

MagnuzBinder committed:
use consistent language: hex => color

mrdoob committed:
ImageBitmapLoader: Latest Firefox no longer complains about options anymore.

mrdoob committed:
Merge pull request #12660 from MagnuzBinder/patch-1

remove duplicate verticesNeedUpdate

mrdoob committed:
Merge pull request #12662 from MagnuzBinder/patch-2

fix typo Mixin => Mixing

mrdoob committed:
Merge pull request #12663 from MagnuzBinder/patch-3

fix index error in example

mrdoob committed:
Merge pull request #12664 from MagnuzBinder/patch-4

fix copy-paste error ParticleSystem => Particle

mrdoob committed:
Merge pull request #12665 from MagnuzBinder/patch-5

Remove unintelligible part of explanation

mrdoob committed:
Merge pull request #12666 from MagnuzBinder/patch-6

consistent naming, hex => color

mrdoob committed:
Merge pull request #12667 from MagnuzBinder/patch-7

fix copy-paste error: models => images

mrdoob committed:
Merge pull request #12668 from MagnuzBinder/patch-8

fix copy-paste error: 2D => 3D

mrdoob committed:
Merge pull request #12669 from MagnuzBinder/patch-9

fix copy-paste error: Vector3 => Box3

mrdoob committed:
Merge pull request #12670 from MagnuzBinder/patch-10

fix copy-paste error: 0, 1 or 2 => 0, 1, 2 or 3

mrdoob committed:
Merge pull request #12671 from MagnuzBinder/patch-11

remove duplicate text: domElement

mrdoob committed:
Merge pull request #12672 from MagnuzBinder/patch-12

fix typo: remove extra dot

mrdoob committed:
Merge pull request #12673 from MagnuzBinder/patch-13

use consistent language: hex => color

mrdoob committed:
Merge pull request #12674 from MagnuzBinder/patch-14

use consistent language: hex => color

mrdoob committed:
Added ImageBitmapLoader to src/ and tweaked *TextureLoader.

mrdoob committed:
Updated builds.

adrs2002 committed:
x file loader update
fix # 12599 unload xfiles.

. Apply structural loading. It corresponds to many X files.
.Separate file of model and animation can be load and bind animation.

mrdoob committed:
Merge pull request #12677 from adrs2002/fix/#12599_xloader_unread_comment

x file loader update

cnspaha committed:
use local path variable

Commit:
GLTFExporter: Export textures as data URIs.

Commit:
GLTFExporter: Include correct mimeType, allow embedImages=false opt-out.

Commit:
GLTFExporter: Only write to canvas if embedImages=true

mrdoob committed:
Merge pull request #12685 from donmccurdy/examples-morphtargets-sphere

Examples: Manually manipulating morph targets on a model.

mrdoob committed:
Updated webgl_Shadowmap_pointlight example.

Commit:
GLTFLoader: Throw unhandled errors.

mrdoob committed:
Improved webgl_shadowmap_pointlight example.

mrdoob committed:
Merge pull request #12687 from donmccurdy/feat-gltf-throw-unhandled-errors

GLTFLoader: Throw unhandled errors.

tmcw committed:
Fix typo and comma splice

mrdoob committed:
Merge pull request #12688 from tmcw/patch-1

Fix typo and comma splice

mrdoob committed:
Merge pull request #12680 from donmccurdy/feat-gltfexporter-texture-data-uris

GLTFExporter: Export textures as data URIs.

mrdoob committed:
Merge pull request #12651 from dmarcos/headObject

Add the ability to specify an alternative Object3D to the camera to apply the pose coming from a VRDevice

mrdoob committed:
Updated builds.

mrdoob committed:
Improved webvr_sandbox example.

mrdoob committed:
Improved webvr_sandbox example.

Commit:
Docs: Example of setURLModifier, IE11 note on GLTFLoader.

WestLangley committed:
Clarification

looeee committed:
FBXLoader_parse_layered_textures

Mugen87 committed:
ImageBitmapLoader: Clean up

mrdoob committed:
Merge pull request #12695 from Mugen87/dev9

ImageBitmapLoader: Clean up

mrdoob committed:
Merge pull request #12690 from WestLangley/dev-cubetex

Docs: CubeTextureLoader Clarification

mrdoob committed:
Merge pull request #12689 from donmccurdy/docs-loading-modifier

Docs: Example of setURLModifier, IE11 note on GLTFLoader.

Mugen87 committed:
Update ImageLoader.html

donmccurdy committed:
GLTFExporter: Handle .flipY on textures.

looeee committed:
FBXLoader correctly parse duplicate maps

looeee committed:
typo

Mugen87 committed:
ImageBitmapLoader: Added documentation

looeee committed:
FBXLoader split monolithic parseScene method

looeee committed:
fix comment indentation

looeee committed:
fix more indentation

looeee committed:
FBXLoader: refactor genGeo

looeee committed:
fix indentation

matb committed:
Propagate error during loading of STL files via onError callback

mrdoob committed:
Merge pull request #12697 from donmccurdy/feat-gltfexporter-flipy

GLTFExporter: Handle .flipY on textures.

matb committed:
Fix styles

donmccurdy committed:
GLTFExporter: Use utf-8 encoding for .glb JSON chunk.

looeee committed:
FBXLoader refactor animation

looeee committed:
removed console log

Mugen87 committed:
Update Texture.html

Mugen87 committed:
Removed ImageUtils

Mugen87 committed:
ImageBitmapLoader: Fix return type

Mugen87 committed:
Docs: Add example to DataTexture

Mugen87 committed:
Docs: Added short explanation of ImageBitmap

mrdoob committed:
Removed unneeded transparent flag in webgl_shadowmap_pointlight example.

mrdoob committed:
Merge pull request #12700 from Mugen87/dev9

ImageBitmapLoader: Added documentation

mrdoob committed:
Merge pull request #12711 from mrdoob/Mugen87-patch-2

Update Texture.html

mrdoob committed:
Merge pull request #12709 from donmccurdy/feat-gltfexporter-utf8

GLTFExporter: Use utf-8 encoding for .glb JSON chunk.

mrdoob committed:
Merge pull request #12696 from mrdoob/Mugen87-patch-1

Update ImageLoader.html

looeee committed:
removed redundant redundant duplicate keyframe check

looeee committed:
removed try catch from addAnimations

looeee committed:
inlined isKeyExistOnFrame

looeee committed:
FBXLoader added support for rotated pivots

looeee committed:
better comments

Commit:
GLTFLoader: Use .byteOffset and .count in sparse accessors.

mrdoob committed:
Merge pull request #12694 from looeee/FBXLoader_layered_textures

FBXLoader: parse layered textures

mrdoob committed:
Merge pull request #12614 from cnspaha/patch-6

ColladaLoader: Added possibility to manually set path

mrdoob committed:
Merge pull request #12701 from looeee/FBXLoader_split_prasescene

Fbx loader split monolithic parseScene method

mrdoob committed:
Merge pull request #12703 from looeee/FBXLoader_gengeom_refactor

FBXLoader: refactor genGeo method

looeee committed:
resolve conflicts

looeee committed:
resolve conflicts

looeee committed:
revert

looeee committed:
resolved conflicts

looeee committed:
added comment

looeee committed:
fixed missing brace

mrdoob committed:
Merge pull request #12699 from looeee/FBXLoader_allow_duplicate_maps

FBXLoader correctly parse duplicate embedded textures

mrdoob committed:
Merge pull request #12717 from looeee/FBXLoader_support_rotated_pivots

FBXLoader support rotated pivots

mrdoob committed:
Merge pull request #12615 from cnspaha/patch-7

TDSLoader: Use path and manager for all Loaders

mrdoob committed:
Merge pull request #12704 from matb/fix-error-handling-stl-loader

Propagate error during loading of STL files via onError callback

looeee committed:
simplify parseAnimations return object

mrdoob committed:
Merge pull request #12712 from Mugen87/dev4

Removed ImageUtils

mrdoob committed:
Merge pull request #12596 from donmccurdy/feat-gltf-sparse-accessors

GLTFLoader: Support sparse accessors.

mrdoob committed:
Merge pull request #12710 from looeee/FBXTree_less_animation_keys

FBXLoader refactor parseAnimation and remove unused entities from return object

mrdoob committed:
Merge pull request #12661 from Mugen87/dev6

ShapeUtils: Use earcut for polygon triangulation (2nd attempt)

mrdoob committed:
Updated builds.

Commit:
Merge branch 'dev' of https://github.com/FishOrBear/three.js into dev

mrdoob commented:
This got messed up...

FishOrBear commented:
@takahirox  Hi,After many years, I continued the discussion of this issue.

https://github.com/mont29/blender-io-fbx/blob/master/io_scene_fbx/parse_fbx.py#L166

I think my conjecture is right, because the code of blender is also like this.

FishOrBear commented:
https://github.com/mrdoob/three.js/blob/dev/examples/js/loaders/FBXLoader.js#L1697

Actually in my file, this property may be null, so I am modifying it so
```
if (node.attrName)
    model.name = THREE.PropertyBinding.sanitizeNodeName(node.attrName);
```
